packages/discord.js/typings/index.d.ts

```diff
@@ -579,6 +565,8 @@ export abstract class CommandInteraction<Cached extends CacheType = CacheType> e
   public reply(
     options: string | MessagePayload | InteractionReplyOptions,
   ): Promise<InteractionResponse<BooleanCache<Cached>>>;
+  public launchActivity(options: LaunchActivityOptions & { withResponse: true }): Promise<InteractionCallbackResponse>;
+  public launchActivity(options?: LaunchActivityOptions): Promise<undefined>;
```

Comment:
`LaunchActivityOptions` may have `withResponse: true` without TypeScript being able to infer that.
This overload should return `InteractionCallbackResponse | undefined`.
You could another overload like the one above, but with `false` returning `undefined`.

The same goes for similar overloads.